### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by dxywill** [titu1994/neural-architecture-search#17]
https://github.com/titu1994/neural-architecture-search/blob/d5f5c9d624026263796984c9b71e4ec6443b61b4/controller.py#L364

HI @titu1994 greak work on the implementation, I just have one question about the loss function.
Based on the REINFORCE algorithm and the original paper, the log likelihood of actions are sampled from m models generated using the same policy network. However, in your implementation it seems to me you only sampled a model and calculated the cross entropy between the raw output logits and the input state. What does that mean?

**Comment by titu1994** [comment]
The paper lacked a lot of implementation details, and therefore I had to make a few calls on how the code was written. I've stated in the readme that this is a best effort version of the paper, and does not follow it closely. 

As to why I used just 1 sampled policy rather than m, it is to preserve computational cost for training. This also makes my implementation much more unstable.

**Comment by dxywill** [comment]
@titu1994 Thanks for your quick reply.
